lib/clean.js

```diff
@@ -23,10 +23,11 @@ function clean(gyp, argv, callback) {
     exists(to_delete, (found) => {
       if (found) {
         if (!gyp.opts.silent_clean) console.log('[' + package_json.name + '] Removing "%s"', to_delete);
-        return rimraf(to_delete).then(
-          (result) => callback(null, result),
```

Comment:
I'm not sure why https://github.com/mapbox/node-pre-gyp/pull/707 added this line. I removed it to return the code to it's original state. @pnappa please check to make sure that I'm not overlooking something

Reply:
The [code](https://github.com/mapbox/node-pre-gyp/blame/8b99dc59ddaac4ed5158e75e70ed485aa08b07ce/lib/clean.js) prior to #707 actually errored. #707 fixed this.

In #720 (this PR) the `try .. catch` clause does the required callback.

This is 👍 

(note: test coverage needs revisit for `clean` as used in `rebuild`)

Reply:
thanks! I've added back the call to `callback`

Reply:
Early return not needed. We return at bottom when no error.

Reply:
`callback` is called with different values in each of the locations

Reply:
The [callback function](https://github.com/mapbox/node-pre-gyp/blob/master/lib/main.js#L81) receives a single argument - the error object. The [pattern](https://github.com/mapbox/node-pre-gyp/blob/master/lib/main.js#L81) used when there is no error is to call without passing a value.

Reply:
Done. I've addressed both comments

My confusion stemmed from the use of the word async, which I usually associate with the version returning a promise rather than the version taking a callback. I think we should be sorted now though